Google.Api.Generator.Tests/ProtoTests/ProtoMsgFake.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Google.Protobuf;
+using Google.Protobuf.Reflection;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Testing
+{
+    public abstract class ProtoMsgFake<T> : IMessage<T> where T : ProtoMsgFake<T>
```

Comment:
I think this may be the only non-generated implementation of `IMessage<T>` I've ever seen...

Reply:
:)